File: Python/ByRound/2160/2160_A_MEX_Partition.py
# Problem: CF 2160 A - MEX Partition
# https://codeforces.com/contest/2160/problem/A

"""
A. MEX Partition
Algorithms/Techniques: Greedy, Set operations, MEX calculation

Time Complexity: O(n^2) in worst case due to checking for MEX in a set,
                 where n is the number of elements in the input array.
Space Complexity: O(n) for storing the set of present elements.

The problem asks to find the minimum score over all valid partitions of a multiset A,
where each partition's multisets must have the same MEX value. The key insight is:
- If we can form a valid partition where each subset has mex = k, then we can try k from 0 upward.
- For any k, if there are at least k distinct elements in A such that we can "split" them into k groups,
  then it's a valid partition with score k.
- The answer is the smallest k for which such a split is possible.

We compute MEX of the full array, and use greedy approach to determine the minimal possible score.
"""
t = int(input())
for i in range(t):
    n = int(input())
    A = list(map(int, input().split()))
    present = set(A)  # Store all unique elements for fast lookup
    mex = 0
    while mex in present:  # Keep incrementing mex until we find one not in the set
        mex += 1
    print(mex)


# https://github.com/VaHiX/CodeForces/

File: Kotlin/ByRound/2160/2160_A_MEX_Partition.kt
// Problem: CF 2160 A - MEX Partition
// https://codeforces.com/contest/2160/problem/A

/*
 * Problem: A. MEX Partition
 * Algorithms/Techniques: Frequency counting, MEX calculation
 *
 * Time Complexity: O(n) per test case, where n is the size of input array.
 * Space Complexity: O(1) since we use a fixed-size array of size 101 for counting.
 *
 * The solution calculates the minimum possible score over all valid partitions of the given multiset.
 * For each test case:
 *   - Count frequency of each number in the range [0, 100].
 *   - Find the smallest non-negative integer (MEX) that is not present in the count.
 * This value represents the minimum score achievable for any valid partition.
 */

fun main() {
    val cin = System.`in`.bufferedReader()
    val cout = System.out.bufferedWriter()
    repeat(cin.readLine().toInt()) {
        val n = cin.readLine().toInt() // Read size of input array
        val a = cin.readLine().split(" ").map { it.toInt() } // Read array elements
        val cnt = IntArray(101) // Count frequency of each number (0 to 100)
        for (v in a) cnt[v]++ // Populate the frequency count
        var ans = 0
        for (mex in 0 until 101) { // Iterate through possible MEX values from 0 upwards
            if (cnt[mex] == 0) { // If current number is not present in input
                ans = mex // This is the MEX, hence minimum score
                break
            }
        }
        cout.write("${ans}\n") // Output result for this test case
    }
    cout.flush() // Ensure output is written
}


// https://github.com/VaHiX/CodeForces/